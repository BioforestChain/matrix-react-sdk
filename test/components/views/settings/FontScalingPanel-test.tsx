/*
Copyright 2021 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

import * as TestUtils from "../../../test-utils";
import FontScalingPanel from "../../../../src/components/views/settings/FontScalingPanel";
import SettingsStore from "../../../../src/settings/SettingsStore";

describe("FontScalingPanel", () => {
    afterEach(() => {
        jest.resetAllMocks();
    });
    it("renders the font scaling UI", () => {
        TestUtils.stubClient();
        const { asFragment } = render(<FontScalingPanel />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("updates the font size", () => {
        jest.spyOn(SettingsStore, "setValue");
        TestUtils.stubClient();
        render(<FontScalingPanel />);

        const slider = screen.getByLabelText("Font size");

        fireEvent.change(slider, { target: { value: 19 } });

        expect(SettingsStore.setValue).toHaveBeenCalledWith("baseFontSizeV2", null, "device", 19);

        expect(screen.getByLabelText("Font size")).toHaveValue("19");
    });
});
