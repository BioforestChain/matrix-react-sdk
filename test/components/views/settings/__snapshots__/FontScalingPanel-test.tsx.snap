// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontScalingPanel renders the font scaling UI 1`] = `
<DocumentFragment>
  <div
    class="mx_SettingsSubsection"
    data-testid="mx_FontScalingPanel"
  >
    <div
      class="mx_SettingsSubsectionHeading"
    >
      <h3
        class="mx_Heading_h4 mx_SettingsSubsectionHeading_heading"
      >
        Font size
      </h3>
    </div>
    <div
      class="mx_SettingsSubsection_content mx_SettingsSubsection_contentStretch"
    >
      <div
        class="mx_FontScalingPanel_preview mx_EventTilePreview_loader"
      >
        <div
          class="mx_Spinner"
        >
          <div
            aria-label="Loadingâ€¦"
            class="mx_Spinner_icon"
            data-testid="spinner"
            role="progressbar"
            style="width: 32px; height: 32px;"
          />
        </div>
      </div>
      <div
        class="mx_FontScalingPanel_fontSlider"
      >
        <div
          class="mx_FontScalingPanel_fontSlider_smallText"
        >
          Aa
        </div>
        <div
          class="mx_Slider"
        >
          <input
            aria-label="Font size"
            autocomplete="off"
            max="21"
            min="11"
            step="1"
            type="range"
            value="16"
          />
          <output
            class="mx_Slider_selection"
            style="left: calc(2px + 50% + 1.2em - 1.2em);"
          >
            <span
              class="mx_Slider_selection_label"
            >
              16
            </span>
          </output>
        </div>
        <div
          class="mx_FontScalingPanel_fontSlider_largeText"
        >
          Aa
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
